### PR TITLE
Add Initial support for DIYRuZ_Flower

### PR DESCRIPTION
**Pull request by Smanar**
It s a DIY device https://github.com/diyruz/flower
This PR create a new custom cluster SOIL_MOISTURE_CLUSTER_ID, and add it too in the XML file
See https://github.com/dresden-elektronik/deconz-rest-plugin/issues/4893

Remark:
It's a DIY device, for advanced user, need firmare update, soldering stuff and ect ...

### CODE REVIEW

**Comment by guny74** [comment]
Any Idea how long this PR will take ?

**Comment by Smanar** [comment]
Lol good question, ATM manup is realy motivated, he is adding new features, but other PR are locked in waiting list.
But now the stable is out, so there is no reason for "massive PR validation" soon.

I have as date 1/2 week for the next 2.13 beta, but not official. On my side I will prefer an intermediate version, too much PR in waiting list

File: general.xml [hunk 1]
```diff
@@ -2890,62 +2890,75 @@ controller device, that supports a keypad and LCD screen.</description>
...
+				<attribute-set id="0x0000" description="Soil Moisture Information">
+					<attribute id="0x0000" name="Measured Value" type="u16" access="r" default="0" required="m"></attribute>
+					<attribute id="0x0001" name="Min Measured Value" type="u16" access="r" required="m" range="0x954d,0x7ffe"></attribute>
+					<attribute id="0x0002" name="Max Measured Value" type="u16" access="r" required="m" range="0x954e,0x7ffe"></attribute>
+					<attribute id="0x0003" name="Tolerance" type="u16" access="r" required="o" range="0x0000,0x0800"></attribute>
+				</attribute-set>
+			</server>
+			<client>
+			</client>
+		</cluster>
```
Comment (manup) [inline general.xml:2905]: Can you please change the `general.xml` to just add the Soil Moisture cluster, but not changing Electrical Measurement cluster intendation. Makes the PR a bit more simple.

**Comment by manup** [comment]
Beside the small intentation thing it looks good. Perfect candidate for DDF later on too :)

File: general.xml (hunk 1 above)
Comment (Smanar) [reply general.xml:2905]: done

**Comment by manup** [comment]
@Smanar the other PR is now merged, can you please adapt this one on top of master?
I think the conflicting lines are in `updateSensorNode()`.

**Comment by Smanar** [comment]
@manup code updated.

**Review by manup — APPROVED** [review]

**Comment by wunni1** [comment]
Hello, I’ve sucessfully paired the DIYRuz_flower Sensor via Phoscon (running on HomeAssistant) 2.14.01 with conbee2. But the sensor doesn’t show the soil moisture. I see only temperature, pressure and humidity. All these values are 0. battery shows up 100%. Gets updated properly. In Deconz I see the cluster soil moisture and when I press the read button or the button on the sensor the values get updated. 
How can I see the moisture in phoscon?

**Comment by Smanar** [comment]
I don't think Phoscon support moisture.
When you press the "read" button you can update temperature and humidity too ?
But strange you miss all values, as you have the GUI can you show how look the node in deconz ?

**Comment by wunni1** [comment]
When I press the read button temp and humid remains unchanged, but I think that’s because of the device hasn’t installed these sensors. Here the picture from deconz GUI
0C6ACFA2-EB41-4A13-8433-13763F864CF9

**Comment by Smanar** [comment]
>but I think that’s because of the device hasn’t installed these sensors

Ha you are using only the moisture ? Because the node look fine.
It s probably because the the device is sleeping, I don't remember if this one have a way to wake up when asking attribute.

what is your deconz version ?

**Comment by wunni1** [comment]
Yes i only want use moisture. Deconz is running v2.14.1.

**Comment by wunni1** [comment]
@Smanar : you wrote in #4893:
"This is another story, you can ask for the support in phoscon when the PR will be validated, but you can too just use a classic third app." 
what do you mean by third app?

**Comment by Smanar** [comment]
One of thoses one for exemple https://phoscon.de/en/conbee2/#hasystems

**Comment by wunni1** [comment]
I tried with nodered and it works. Soil moisture gets updated very well….

**Comment by ma-gu-16** [comment]
its still not working version: 2.24.2
@ iobroker
@ conbee II

**Pull request merged**